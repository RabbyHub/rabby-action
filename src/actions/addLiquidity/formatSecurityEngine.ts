import {
  AddLiquidityRequireData,
  FormatSecurityEngineContext,
} from '../../types';

export const formatSecurityEngineAddLiquidity: FormatSecurityEngineContext =
  async (options) => {
    const { actionData, requireData, chainId } = options;

    if (!actionData.addLiquidity || !chainId) {
      return {} as any;
    }

    const data = requireData as AddLiquidityRequireData;

    return {
      addLiquidity: {
        receiverMatch:
          data.receiver?.toLowerCase() === data.sender.toLowerCase(),
        diff: data.diff,
      },
    };
  };
