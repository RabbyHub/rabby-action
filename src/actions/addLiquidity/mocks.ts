export const preExecData = {
  abi: {
    func: 'mint',
    params: [
      [
        '0x18Bc5bcC660cf2B9cE3cd51a404aFe1a0cBD3C22',
        '0x833589fCD6eDb6E08f4c7C32D4f71b54bdA02913',
        10,
        -4970,
        -4960,
        2955759909,
        885199302,
        2940981110,
        880773306,
        '0xE2866F2291B5c5bafD2858B9247f93745108A523',
        1754366820,
        0,
      ],
    ],
  },
  abi_str:
    '{\n    "func": "mint",\n    "params": [\n        [\n            "0x18Bc5bcC660cf2B9cE3cd51a404aFe1a0cBD3C22",\n            "0x833589fCD6eDb6E08f4c7C32D4f71b54bdA02913",\n            10,\n            -4970,\n            -4960,\n            2955759909,\n            885199302,\n            2940981110,\n            880773306,\n            "0xE2866F2291B5c5bafD2858B9247f93745108A523",\n            1754366820,\n            0\n        ]\n    ]\n}',
  balance_change: {
    success: false,
    error: {
      code: 1002,
      msg: 'execution reverted: execution reverted',
    },
    send_token_list: [],
    receive_token_list: [],
    send_nft_list: [],
    receive_nft_list: [],
    usd_value_change: 0,
  },
  gas: {
    success: true,
    error: null,
    gas_used: 29320,
    gas_limit: 0,
    gas_ratio: 1.3,
  },
  is_gnosis: false,
  native_token: {
    id: 'base',
    chain: 'base',
    name: 'ETH',
    symbol: 'ETH',
    display_symbol: null,
    optimized_symbol: 'ETH',
    decimals: 18,
    logo_url:
      'https://static.debank.com/image/coin/logo_url/eth/6443cdccced33e204d90cb723c632917.png',
    protocol_id: '',
    price: 3629.55,
    price_24h_change: 0.018385589869839923,
    credit_score: 46503412.60907772,
    is_verified: true,
    is_scam: false,
    is_suspicious: null,
    is_core: true,
    is_wallet: true,
    time_at: null,
    low_credit_score: false,
    amount: 1e-5,
  },
  pre_exec_version: 'v2',
  pre_exec: {
    success: false,
    error: {
      code: 1002,
      msg: 'execution reverted: execution reverted',
    },
  },
  type_call: {
    action: 'mint',
    contract: '0x827922686190790b37229fd06084350e74485b72',
    contract_protocol_name: 'Aerodrome V3',
    contract_protocol_logo_url:
      'https://static.debank.com/image/project/logo_url/base_aerodrome/f02d753bc321dc8ba480f0424a686482.png',
  },
  trace_id: '663ae6c6c0ac4b1aa5651fadba316de2',
} as any;

export const parseTxData = {
  action: { type: 'cancel_tx' },
  contract_call: null,
  log_id: 441860918,
} as any;

export const txData = {
  chainId: 42161,
  from: '0x4686fd7c0bc5d06346d416808916ba1dbd3f2336',
  to: '0x4686fd7c0bc5d06346d416808916ba1dbd3f2336',
  value: '0x0',
  data: '0x',
  gas: '0x0',
  maxFeePerGas: '0xb71b00',
  maxPriorityFeePerGas: '0xb71b00',
  nonce: '0x4',
};
