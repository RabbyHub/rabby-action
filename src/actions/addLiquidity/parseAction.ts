import { ParseAction } from '../../types';
import { TokenItem } from '@rabby-wallet/rabby-api/dist/types';

interface AddLiquidityActionData {
  receiver: string;
  exchange_rate: string;
  min_exchange_rate: string;
  max_exchange_rate: string;
  token0: TokenItem;
  token1: TokenItem;
}

export const parseActionAddLiquidity: ParseAction = (options) => {
  const { data } = options;

  if (data?.type !== 'add_liquidity') {
    return {};
  }

  const {
    receiver,
    exchange_rate,
    min_exchange_rate,
    max_exchange_rate,
    token0,
    token1,
  } = data.data as unknown as AddLiquidityActionData;

  return {
    addLiquidity: {
      receiver,
      exchange_rate,
      min_exchange_rate,
      max_exchange_rate,
      token0,
      token1,
    },
  };
};
