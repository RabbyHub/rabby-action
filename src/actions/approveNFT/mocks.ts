export const preExecData = {
  abi: {
    func: 'approve',
    params: ['0x7a250d5630B4cF539739dF2C5dAcb4c659F2488D', 745],
  },
  abi_str:
    '{\n    "func": "approve",\n    "params": [\n        "0x7a250d5630B4cF539739dF2C5dAcb4c659F2488D",\n        745\n    ]\n}',
  balance_change: {
    success: true,
    error: null,
    send_token_list: [],
    receive_token_list: [],
    send_nft_list: [],
    receive_nft_list: [],
    usd_value_change: 0,
  },
  gas: {
    success: true,
    error: null,
    gas_used: 48759,
    gas_limit: 49146,
    gas_ratio: 1.3,
  },
  is_gnosis: false,
  native_token: {
    id: 'eth',
    chain: 'eth',
    name: 'ETH',
    symbol: 'ETH',
    display_symbol: null,
    optimized_symbol: 'ETH',
    decimals: 18,
    logo_url:
      'https://static.debank.com/image/coin/logo_url/eth/6443cdccced33e204d90cb723c632917.png',
    protocol_id: '',
    price: 2632.17,
    price_24h_change: 0.020513635693958674,
    credit_score: 100000000.0,
    is_verified: true,
    is_scam: false,
    is_suspicious: false,
    is_core: true,
    is_wallet: true,
    time_at: 1483200000.0,
    amount: 0.00147682999416643,
  },
  pre_exec_version: 'v2',
  pre_exec: {
    success: true,
    error: null,
  },
  type_single_nft_approval: {
    spender: '0x7a250d5630b4cf539739df2c5dacb4c659f2488d',
    spender_protocol_name: 'Uniswap V2',
    spender_protocol_logo_url:
      'https://static.debank.com/image/project/logo_url/uniswap2/87a541b3b83b041c8d12119e5a0d19f0.png',
    token_symbol: 'Metascapes #745',
    token_amount: 1,
    is_infinity: false,
    is_nft: true,
    nft: {
      id: '12ce1dbcf984dd9174065edc9531a09b',
      contract_id: '0x75d639e5e52b4ea5426f2fb46959b9c3099b729a',
      inner_id: '745',
      chain: 'eth',
      symbol: 'MTSCPS',
      name: 'Metascapes #745',
      description:
        '2555 hand-picked AI landscape images of a unique collaboration between Cath Simard, Ryan Newburn, Iurie Belegurschi and Artificial Intelligence, bridging the natural and the supernatural.\n\nhttps://metascapes.sloika.xyz/',
      content_type: 'image_url',
      content:
        'https://static.debank.com/image/eth_nft/local_url/12ce1dbcf984dd9174065edc9531a09b/47202001ef73e7586c53412c3bc72629.png',
      thumbnail_url:
        'https://static.debank.com/image/eth_nft/thumbnail_url/12ce1dbcf984dd9174065edc9531a09b/e5a971f66038f0159ac918b0faacd49e.png',
      total_supply: 1,
      attributes: [
        {
          trait_type: 'Type',
          value: 'Image',
        },
      ],
      detail_url:
        'https://opensea.io/assets/0x75d639e5e52b4ea5426f2fb46959b9c3099b729a/745',
      collection_id: 'eth:0x75d639e5e52b4ea5426f2fb46959b9c3099b729a',
      is_erc1155: false,
      is_erc721: true,
      pay_token: {
        id: 'eth',
        amount: 0.099,
        time_at: 1678369547.0,
        date_at: '2023-03-09',
      },
      usd_price: 151.69176000000002,
      collection: {
        id: 'eth:0x75d639e5e52b4ea5426f2fb46959b9c3099b729a',
        chain: 'eth',
        name: 'The Metascapes',
        description: null,
        logo_url:
          'https://static.debank.com/image/nft_collection/logo_url/the-metascapes/065c2cd129c3c0023c6cb3ea2051be4a.png',
        is_verified: true,
        credit_score: 216.306447552488,
        is_scam: false,
        is_suspicious: false,
        is_core: true,
        floor_price: 0.0243,
      },
    },
    token: {
      raw_amount: 1,
      raw_amount_hex_str: '0x1',
      is_infinity: false,
    },
  },
  trace_id: '182385710d964fdaba026f2a76a6dc36',
} as any;

export const parseTxData = {
  action: {
    type: 'approve_nft',
    data: {
      spender: '0x7a250d5630b4cf539739df2c5dacb4c659f2488d',
      nft: {
        id: '12ce1dbcf984dd9174065edc9531a09b',
        contract_id: '0x75d639e5e52b4ea5426f2fb46959b9c3099b729a',
        inner_id: '745',
        chain: 'eth',
        symbol: 'MTSCPS',
        name: 'Metascapes #745',
        description:
          '2555 hand-picked AI landscape images of a unique collaboration between Cath Simard, Ryan Newburn, Iurie Belegurschi and Artificial Intelligence, bridging the natural and the supernatural.\n\nhttps://metascapes.sloika.xyz/',
        content_type: 'image_url',
        content:
          'https://static.debank.com/image/eth_nft/local_url/12ce1dbcf984dd9174065edc9531a09b/47202001ef73e7586c53412c3bc72629.png',
        thumbnail_url:
          'https://static.debank.com/image/eth_nft/thumbnail_url/12ce1dbcf984dd9174065edc9531a09b/e5a971f66038f0159ac918b0faacd49e.png',
        total_supply: 1,
        attributes: [
          {
            trait_type: 'Type',
            value: 'Image',
          },
        ],
        detail_url:
          'https://opensea.io/assets/0x75d639e5e52b4ea5426f2fb46959b9c3099b729a/745',
        collection_id: 'eth:0x75d639e5e52b4ea5426f2fb46959b9c3099b729a',
        is_erc1155: false,
        is_erc721: true,
        pay_token: {
          id: 'eth',
          amount: 0.099,
          time_at: 1678369547.0,
          date_at: '2023-03-09',
        },
        usd_price: 151.69176000000002,
        collection: {
          id: 'eth:0x75d639e5e52b4ea5426f2fb46959b9c3099b729a',
          chain: 'eth',
          name: 'The Metascapes',
          description: null,
          logo_url:
            'https://static.debank.com/image/nft_collection/logo_url/the-metascapes/065c2cd129c3c0023c6cb3ea2051be4a.png',
          is_verified: true,
          credit_score: 216.306447552488,
          is_scam: false,
          is_suspicious: false,
          is_core: true,
          floor_price: 0.0243,
        },
      },
    },
  },
  contract_call: {
    func: 'approve',
    contract: {
      id: '0x75d639e5e52b4ea5426f2fb46959b9c3099b729a',
      protocol: null,
    },
  },
  log_id: 84523037,
} as any;

export const txData = {
  chainId: 1,
  data: '0x095ea7b30000000000000000000000007a250d5630b4cf539739df2c5dacb4c659f2488d00000000000000000000000000000000000000000000000000000000000002e9',
  from: '0x5853ed4f26a3fcea565b3fbc698bb19cdf6deb85',
  gas: '',
  maxFeePerGas: '0x4190ab00',
  maxPriorityFeePerGas: '0x4190ab00',
  nonce: '0x168f',
  to: '0x75d639e5e52b4ea5426f2fb46959b9c3099b729a',
  value: '0x0',
};
