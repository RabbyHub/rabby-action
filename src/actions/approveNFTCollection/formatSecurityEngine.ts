import {
  ApproveNFTRequireData,
  FormatSecurityEngineContext,
} from '../../types';

export const formatSecurityEngineApproveNFTCollection: FormatSecurityEngineContext =
  async (options) => {
    const { actionData, requireData, chainId, provider } = options;

    if (!actionData.approveNFTCollection || !chainId) {
      return {};
    }
    const data = requireData as ApproveNFTRequireData;
    const { spender } = actionData.approveNFTCollection;
    return {
      collectionApprove: {
        chainId,
        spender,
        isEOA: data.isEOA,
        riskExposure: data.riskExposure,
        deployDays: provider.getTimeSpan(
          Math.floor(Date.now() / 1000) - data.bornAt
        ).d,
        isDanger: !!data.isDanger,
      },
    };
  };
