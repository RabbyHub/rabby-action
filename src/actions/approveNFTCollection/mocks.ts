export const preExecData = {
  abi: {
    func: 'setApprovalForAll',
    params: ['0x7a250d5630B4cF539739dF2C5dAcb4c659F2488D', true],
  },
  abi_str:
    '{\n    "func": "setApprovalForAll",\n    "params": [\n        "0x7a250d5630B4cF539739dF2C5dAcb4c659F2488D",\n        true\n    ]\n}',
  balance_change: {
    success: true,
    error: null,
    send_token_list: [],
    receive_token_list: [],
    send_nft_list: [],
    receive_nft_list: [],
    usd_value_change: 0,
  },
  gas: {
    success: true,
    error: null,
    gas_used: 46210,
    gas_limit: 46592,
    gas_ratio: 1.3,
  },
  is_gnosis: false,
  native_token: {
    id: 'eth',
    chain: 'eth',
    name: 'ETH',
    symbol: 'ETH',
    display_symbol: null,
    optimized_symbol: 'ETH',
    decimals: 18,
    logo_url:
      'https://static.debank.com/image/coin/logo_url/eth/6443cdccced33e204d90cb723c632917.png',
    protocol_id: '',
    price: 2640.55,
    price_24h_change: 0.006898103307581486,
    credit_score: 100000000.0,
    is_verified: true,
    is_scam: false,
    is_suspicious: false,
    is_core: true,
    is_wallet: true,
    time_at: 1483200000.0,
    amount: 0.00147682999416643,
  },
  pre_exec_version: 'v2',
  pre_exec: {
    success: true,
    error: null,
  },
  type_nft_collection_approval: {
    spender: '0x7a250d5630b4cf539739df2c5dacb4c659f2488d',
    spender_protocol_name: 'Uniswap V2',
    spender_protocol_logo_url:
      'https://static.debank.com/image/project/logo_url/uniswap2/87a541b3b83b041c8d12119e5a0d19f0.png',
    token_symbol: 'Metascapes',
    is_nft: true,
    nft_contract: {
      id: '0x75d639e5e52b4ea5426f2fb46959b9c3099b729a',
      chain: 'eth',
      name: 'Metascapes',
      symbol: 'MTSCPS',
      is_core: true,
      time_at: 1643633806.0,
      collection: {
        id: 'eth:0x75d639e5e52b4ea5426f2fb46959b9c3099b729a',
        chain: 'eth',
        name: 'The Metascapes',
        description: null,
        logo_url:
          'https://static.debank.com/image/nft_collection/logo_url/the-metascapes/065c2cd129c3c0023c6cb3ea2051be4a.png',
        is_verified: true,
        credit_score: 206.063934578252,
        is_scam: false,
        is_suspicious: false,
        is_core: true,
        floor_price: 0.0242,
      },
    },
    token: {
      raw_amount: 1,
      raw_amount_hex_str: '0x1',
      is_infinity: false,
    },
    token_amount: 1,
    is_infinity: false,
  },
  trace_id: 'b6fc140e6eca47af87ee1cf255a521c2',
} as any;

export const parseTxData = {
  action: {
    type: 'approve_collection',
    data: {
      spender: '0x7a250d5630b4cf539739df2c5dacb4c659f2488d',
      collection: {
        chain_id: 'eth',
        id: '0x75d639e5e52b4ea5426f2fb46959b9c3099b729a',
        name: 'The Metascapes',
        description: null,
        logo_url:
          'https://static.debank.com/image/nft_collection/logo_url/the-metascapes/065c2cd129c3c0023c6cb3ea2051be4a.png',
        is_verified: null,
        is_core: true,
        is_visible: true,
        rank_at: null,
        nft_count: 2555,
        holder_count: 1328,
        buy_holder_count: 511,
        buy_holder_usd_value: 14098606.89417298,
        total_est_usd_value: 162518.21019654564,
        total_est_value: 0.0,
        thirdparty: {},
        avg_price_24h: 0.014914361573834049,
        avg_price_last_24h: 0.014917500000000002,
        max_price_24h: 0.014914361573834049,
        max_price_last_24h: 0.014917500000000002,
        volume_24h: 0.0,
        volume_last_24h: 0.0,
        credit_score: 206.063934578252,
        credit_score_rank_at: 2653,
        floor_price: 0.0242,
        floor_price_24h: 0.0,
        is_scam: false,
      },
    },
    is_asset_changed: false,
    is_involving_privacy: false,
  },
  contract_call: {
    func: 'setApprovalForAll',
    contract: {
      id: '0x75d639e5e52b4ea5426f2fb46959b9c3099b729a',
      protocol: null,
    },
  },
  log_id: 85209228,
} as any;

export const txData = {
  chainId: 1,
  from: '0x5853ed4f26a3fcea565b3fbc698bb19cdf6deb85',
  to: '0x75d639e5e52b4ea5426f2fb46959b9c3099b729a',
  value: '0x0',
  data: '0xa22cb4650000000000000000000000007a250d5630b4cf539739df2c5dacb4c659f2488d0000000000000000000000000000000000000000000000000000000000000001',
  gas: '',
  maxFeePerGas: '0x4190ab00',
  maxPriorityFeePerGas: '0x4190ab00',
  nonce: '0x168f',
};
