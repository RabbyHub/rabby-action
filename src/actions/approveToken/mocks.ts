export const preExecData = {
  abi: {
    func: 'approve',
    params: [
      '0x7a250d5630B4cF539739dF2C5dAcb4c659F2488D',
      100000000000000000000,
    ],
  },
  abi_str:
    '{\n    "func": "approve",\n    "params": [\n        "0x7a250d5630B4cF539739dF2C5dAcb4c659F2488D",\n        100000000000000000000\n    ]\n}',
  balance_change: {
    success: true,
    error: null,
    send_token_list: [],
    receive_token_list: [],
    send_nft_list: [],
    receive_nft_list: [],
    usd_value_change: 0,
  },
  gas: {
    success: true,
    error: null,
    gas_used: 46158,
    gas_limit: 46540,
    gas_ratio: 1.3,
  },
  is_gnosis: false,
  native_token: {
    id: 'eth',
    chain: 'eth',
    name: 'ETH',
    symbol: 'ETH',
    display_symbol: null,
    optimized_symbol: 'ETH',
    decimals: 18,
    logo_url:
      'https://static.debank.com/image/coin/logo_url/eth/6443cdccced33e204d90cb723c632917.png',
    protocol_id: '',
    price: 2640.38,
    price_24h_change: 0.009809808278483855,
    credit_score: 100000000.0,
    is_verified: true,
    is_scam: false,
    is_suspicious: false,
    is_core: true,
    is_wallet: true,
    time_at: 1483200000.0,
    amount: 0.00336741278341931,
  },
  pre_exec_version: 'v2',
  pre_exec: {
    success: true,
    error: null,
  },
  type_token_approval: {
    spender: '0x7a250d5630b4cf539739df2c5dacb4c659f2488d',
    spender_protocol_name: 'Uniswap V2',
    spender_protocol_logo_url:
      'https://static.debank.com/image/project/logo_url/uniswap2/87a541b3b83b041c8d12119e5a0d19f0.png',
    token_symbol: 'DAI',
    token_amount: 100.0,
    is_infinity: false,
    is_nft: false,
    nft: null,
    token: {
      id: '0x6b175474e89094c44da98b954eedeac495271d0f',
      chain: 'eth',
      name: 'Dai Stablecoin',
      symbol: 'DAI',
      display_symbol: '',
      optimized_symbol: 'DAI',
      decimals: 18,
      logo_url:
        'https://static.debank.com/image/eth_token/logo_url/0x6b175474e89094c44da98b954eedeac495271d0f/549c4205dbb199f1b8b03af783f35e71.png',
      protocol_id: 'makerdao',
      price: 0.9999,
      price_24h_change: 0.0,
      credit_score: 4768525.856018129,
      is_verified: true,
      is_scam: false,
      is_suspicious: false,
      is_core: true,
      is_wallet: true,
      time_at: 1573672677.0,
      raw_amount: 100000000000000000000,
      raw_amount_hex_str: '0x56bc75e2d63100000',
      is_infinity: false,
    },
  },
  trace_id: '64ef985fd2c84786a171fec68a08fe8f',
} as any;

export const parseTxData = {
  action: {
    type: 'approve_token',
    data: {
      spender: '0x7a250d5630b4cf539739df2c5dacb4c659f2488d',
      token: {
        id: '0x6b175474e89094c44da98b954eedeac495271d0f',
        chain: 'eth',
        name: 'Dai Stablecoin',
        symbol: 'DAI',
        display_symbol: '',
        optimized_symbol: 'DAI',
        decimals: 18,
        logo_url:
          'https://static.debank.com/image/eth_token/logo_url/0x6b175474e89094c44da98b954eedeac495271d0f/549c4205dbb199f1b8b03af783f35e71.png',
        protocol_id: 'makerdao',
        price: 0.9999,
        price_24h_change: 0.0,
        credit_score: 4768525.856018129,
        is_verified: true,
        is_scam: false,
        is_suspicious: false,
        is_core: true,
        is_wallet: true,
        time_at: 1573672677.0,
        amount: 100.0,
        raw_amount: '100000000000000000000',
      },
    },
  },
  contract_call: {
    func: 'approve',
    contract: {
      id: '0x6b175474e89094c44da98b954eedeac495271d0f',
      protocol: {
        name: 'Maker',
        logo_url:
          'https://static.debank.com/image/project/logo_url/makerdao/3821328c7c6d5ac4fc87e2c2e4d1c684.png',
      },
    },
  },
  log_id: 85163324,
} as any;

export const txData = {
  chainId: 1,
  from: '0x5853ed4f26a3fcea565b3fbc698bb19cdf6deb85',
  to: '0x6b175474e89094c44da98b954eedeac495271d0f',
  value: '0x0',
  data: '0x095ea7b30000000000000000000000007a250d5630b4cf539739df2c5dacb4c659f2488d0000000000000000000000000000000000000000000000056bc75e2d63100000',
  gas: '',
  maxFeePerGas: '0x47868c00',
  maxPriorityFeePerGas: '0x47868c00',
  nonce: '0x1691',
};
