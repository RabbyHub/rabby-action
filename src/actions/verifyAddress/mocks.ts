export const parseTxData = {
  action: {
    type: 'verify_address',
    data: {
      user: '0x5853ed4f26a3fcea565b3fbc698bb19cdf6deb85',
      allow_origins: ['https://metamask.github.io'],
      protocol: null,
      desc: 'You are verifying address on metamask.github.io',
    },
    is_asset_changed: false,
    is_involving_privacy: true,
  },
  log_id: 23739932,
} as any;

export const text = `metamask.github.io wants you to sign in with your Ethereum account:
0x5853ed4f26a3fcea565b3fbc698bb19cdf6deb85

I accept the MetaMask Terms of Service: https://community.metamask.io/tos

URI: https://metamask.github.io
Version: 1
Chain ID: 1
Nonce: 32891757
Issued At: 2021-09-30T16:25:24.000Z`;
