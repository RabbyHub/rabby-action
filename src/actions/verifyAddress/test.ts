import { SENDER, ORIGIN, formatProvider } from '../../../__mocks__';
import { parseTxData, text } from './mocks';
import { parseActionVerifyAddress } from './parseAction';
import { formatSecurityEngineVerifyAddress } from './formatSecurityEngine';
import { parseTextAction } from '../../utils/parseTextAction';
import { parseAction, formatSecurityEngineContext } from '../..';

/**
 * https://metamask.github.io/test-dapp/#siwe
 * Sign In With Ethereum
 * - [Sign In With Ethereum] button
 */
test.each([
  [
    parseTextAction(parseActionVerifyAddress),
    formatSecurityEngineVerifyAddress,
  ],
  [parseAction, formatSecurityEngineContext],
])('VerifyAddress -> Text', async (_parseAction, _format) => {
  const actionData = _parseAction({
    type: 'text',
    data: parseTxData['action'],
    text,
    sender: SENDER,
  });
  expect(actionData).toMatchSnapshot('parseActionVerifyAddress');

  const ctx = await _format({
    type: 'text',
    actionData,
    origin: ORIGIN,
    provider: formatProvider,
    isTestnet: false,
    chainId: 'eth',
    requireData: null,
  });
  expect(ctx).toMatchSnapshot('formatSecurityEngineVerifyAddress');
});
