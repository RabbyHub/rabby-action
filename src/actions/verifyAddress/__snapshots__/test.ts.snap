// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`VerifyAddress -> Text: formatSecurityEngineVerifyAddress 1`] = `
{
  "verifyAddress": {
    "allowOrigins": [
      "https://metamask.github.io",
    ],
    "origin": "https://rabby.io",
  },
}
`;

exports[`VerifyAddress -> Text: formatSecurityEngineVerifyAddress 2`] = `
{
  "verifyAddress": {
    "allowOrigins": [
      "https://metamask.github.io",
    ],
    "origin": "https://rabby.io",
  },
}
`;

exports[`VerifyAddress -> Text: parseActionVerifyAddress 1`] = `
{
  "sender": "0x5853ed4f26a3fcea565b3fbc698bb19cdf6deb85",
  "verifyAddress": {
    "allow_origins": [
      "https://metamask.github.io",
    ],
    "desc": "You are verifying address on metamask.github.io",
    "protocol": null,
    "user": "0x5853ed4f26a3fcea565b3fbc698bb19cdf6deb85",
  },
}
`;

exports[`VerifyAddress -> Text: parseActionVerifyAddress 2`] = `
{
  "sender": "0x5853ed4f26a3fcea565b3fbc698bb19cdf6deb85",
  "verifyAddress": {
    "allow_origins": [
      "https://metamask.github.io",
    ],
    "desc": "You are verifying address on metamask.github.io",
    "protocol": null,
    "user": "0x5853ed4f26a3fcea565b3fbc698bb19cdf6deb85",
  },
}
`;
